tests/test_controller.py

```diff
@@ -22,9 +22,28 @@
 from ansible_rulebook.exception import (
     ControllerApiException,
     JobTemplateNotFoundException,
+    WorkflowJobTemplateNotFoundException,
 )
 from ansible_rulebook.job_template_runner import job_template_runner
 
+from .data.awx_test_data import (
+    JOB_1_RUNNING,
+    JOB_1_SLUG,
+    JOB_1_SUCCESSFUL,
+    JOB_TEMPLATE_1_LAUNCH_SLUG,
+    JOB_TEMPLATE_2_LAUNCH_SLUG,
+    JOB_TEMPLATE_NAME_1,
+    JOB_TEMPLATE_POST_RESPONSE,
+    NO_JOB_TEMPLATE_PAGE1_RESPONSE,
+    ORGANIZATION_NAME,
+    UNIFIED_JOB_TEMPLATE_PAGE1_RESPONSE,
+    UNIFIED_JOB_TEMPLATE_PAGE1_SLUG,
+    UNIFIED_JOB_TEMPLATE_PAGE2_RESPONSE,
+    UNIFIED_JOB_TEMPLATE_PAGE2_SLUG,
+)
+
+os.environ["EDA_JOB_TEMPLATE_REFRESH_DELAY"] = "0.05"
```

Comment:
This is not a good practice because you are changing the environment for the whole process and eventually generate unexpected side effects. It would be preferably a fixture. Ideally use that fixture for all these tests that need it using "userfixtures"

def sometest():

Reply:
@Alex-Izquierdo I had this delayed setting on a test by test basis, the problem is in how the Job_template_runner instance is created. There is a single instance for the entire run and it gets loaded here. https://github.com/ansible/ansible-rulebook/blob/03b5fe4ed064f4c9b06631b5215e2f3286600101/tests/test_controller.py#L26 the env var is accessed once in the constructor. https://github.com/ansible/ansible-rulebook/blob/03b5fe4ed064f4c9b06631b5215e2f3286600101/ansible_rulebook/job_template_runner.py#L46 I am not sure how we would overwrite it. Either we would have to check for this env_var for every get/post call or add an argument to pass in the refresh_delay

Reply:
Why not modify the attribute after instantiaton as you do with other attributes in the tests? https://github.com/ansible/ansible-rulebook/pull/543/files#diff-f7dbf2e52de217bf88ebb0364a5529e8f303b44ed043c4c3e4a1a1d681588ea4R116
In fact, you don't need to do it for each test because you are modifying the global instance. Or even better, you could create a fixture for it: 

```
@pytest.fixture
def mocked_job_template_runner():
    from ansible_rulebook.job_template_runner import job_template_runner
    job_template_runner.host = "https://example.com"
    job_template_runner.token = "DUMMY"
    job_template_runner.refresh_delay = 0.5
    return job_template_runner
```